wolfcrypt/src/dh.c

```diff
@@ -1522,6 +1522,46 @@ int wc_DhCheckPubKey(DhKey* key, const byte* pub, word32 pubSz)
 }
 
 
+/**
+ * Quick validity check of public key value agaist prime.
+ * Checks are:
+ *   - Public key not 0 or 1
+ *   - Public key not equal to prime or prime - 1
+ *   - Public key not bigger than prime.
+ *
+ * prime    Big-endian encoding of prime in bytes.
+ * primeSz  Size of prime in bytes.
+ * pub      Big-endian encoding of public key in bytes.
+ * pubSz    Size of public key in bytes.
+ */
+int wc_DhCheckPubValue(const byte* prime, word32 primeSz, const byte* pub,
```

Comment:
Are these checks redundant if wc_DhCehckPubKey is being used?

Reply:
wc_DhCheckPubKey doesn't detect all the issues.
Some errors were not reported until the finish message as I remember.
Also, these are quick to do and saves the CPU usage.